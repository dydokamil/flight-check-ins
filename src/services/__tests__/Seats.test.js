import React from "react"
import Adapter from "enzyme-adapter-react-16"
import Enzyme, { mount, shallow } from "enzyme"
import { MemoryRouter } from "react-router-dom"
import ReactDOM from "react-dom"
import renderer from "react-test-renderer"
import { Link } from "react-router-dom"

import {
  mapStateToProps,
  mapDispatchToProps,
  Seats,
} from "../../components/Seats"

Enzyme.configure({ adapter: new Adapter() })

describe("test Seats component", () => {
  it("should render the component", () => {
    const component = shallow(
      <Seats fetchSeats={jest.fn()} seatsReducer={{ basePrice: 0 }} />,
    )

    expect(component).toMatchSnapshot()
  })
  it("should assign state from props", () => {
    const INITIAL_STATE = {
      seatsReducer: {},
      sessionReducer: { token: "123abc" },
      reservationReducer: { error: "Some error" },
    }
    expect(mapStateToProps(INITIAL_STATE)).toEqual({
      seatsReducer: INITIAL_STATE.seatsReducer,
      token: INITIAL_STATE.sessionReducer.token,
      reservationError: INITIAL_STATE.reservationReducer.error,
    })
  })

  it("should call `fetchSeats` when page is loaded", () => {
    const dispatch = jest.fn()
    mapDispatchToProps(dispatch).fetchSeats()
    expect(dispatch).toHaveBeenCalledTimes(1)
  })

  it("should call `onRandomReservation` when page is loaded", () => {
    const dispatch = jest.fn()
    mapDispatchToProps(dispatch).onRandomReservation({})
    expect(dispatch).toHaveBeenCalledTimes(1)
  })
})
