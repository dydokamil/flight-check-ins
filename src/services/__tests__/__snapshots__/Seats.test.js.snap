// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Seats component should render the component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Seats
    fetchSeats={[Function]}
    seatsReducer={
        Object {
            "basePrice": 0,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "No seats.",
      "tag": "div",
    },
    "ref": null,
    "rendered": "No seats.",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "No seats.",
        "tag": "div",
      },
      "ref": null,
      "rendered": "No seats.",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
